index.js

```diff
@@ -1,20 +1,302 @@
+    "gas can",
+    "An old rusted can of gas. It should be light enough to carry.",
+    true
+  );
+
+  //Rooms objects
+
+  let boat = new Room(
+    "boat",
+    "An old fishing boat, worn but in working condition. There are keys in the ignition. It feels familiar, but you are sure you have never been here before.",
+    ["boatKey"],
+    ["beach1"],
+    false
+  );
+
+  let beach1 = new Room(
+    "beach1",
+    "On this desolate patch of sand, there is a large, decrepit lighthouse on a small peninsula that juts into the ocean to the west . The door is sturdy and the handle is old and rusted. The cliffs to the east remain. The mysterious boat remains to the north. The beach continues to the south.\n\n",
+    [],
+    ["boat", "lighthouse", "beach2"],
+    false
+  );
+
+  let lighthouse = new Room(
+    "lighthouse",
+    "The heavy door of the lighthouse creeks open. Beams of light shine in through the windows, catching the dust suspended in the air. The stairs that circle the inside walls of the lighthouse are crumbled, leaving the upper levels inaccessible. You wouldn’t climb them if you could. The smell of gasoline wafts from a large can in the room.\n\n",
+    ["gasCan"],
+    ["beach1"],
+    true
+  );
+
+  let beach2 = new Room(
+    "beach2",
+    "The end of the beach. The sea stretches endlessly to the south and west. To your east, you see a small cave in the base of the cliff. The lighthouse towers to the northwest.\n\n",
+    [],
+    ["beach1", "cave1"],
+    false
+  );
+
+  let cave1 = new Room(
+    "cave1",
+    "The small cave entrance opens up into a large chamber, lit by a torch on the wall. Snakes and spiders slither and scurry into the shadows as you enter. There appears to be another chamber in the cave to your south. The air is thick and still.\n\n",
+    ["cave2", "beach2"],
+    false
+  );
+
+  let cave2 = new Room(
+    "cave2",
+    "There is barely enough light in the room to see a figure slumped against the far wall of the chamber. It is a skeleton. Only bones and tattered clothes remain.\n\n",
+    [lighthouseKey, letter, skeleton],
+    ["cave1"],
+    false
+  );
+
+  /* below is the table that dictates what rooms each room can lead to*/
+  let transitions = {
+    boat: ["beach1"],
+    beach1: ["lighthouse", "beach2", "boat"],
+    lighthouse: ["beach2"],
+    beach2: ["beach1", "cave1"],
+    cave1: ["beach2", "cave2"],
+    cave2: ["cave1"],
+  };
+
+  /*function for moving from room to room*/
+  function changeRoom(nextRoom) {
+    if (transitions[currentRoom].includes(nextRoom)) {
+      currentRoom = nextRoom;
+    } else {
+      console.log(`You can\'t get there from here`);
+    }
+  }
+
+  /*function for picking up items. does not work currently. will be adjusted and resubmitted*/
+  function take() {
+    if (this.moveable) {
+      player.playerInv.push(this.item);
+      console.log("You've taken the " + this.item + ".");
+    } else console.log("You can't take that!");
+  }
```

Comment:
This looks like it was on the right track, you are able to access the description of the key while in the "boat room" - However I also noticed you have accomplished the same goal on lines 259-272 outside of this dedicated function. This function could be removed completely, or set up to do the same logic you have currently working for the gas can and skeleton keys, at which point you could just call this function when needed.